### PR TITLE
[Scope change] add is warranty case to form

### PR DESCRIPTION
**Pull request by Gustav-Eikaas**
# Description

Please include a summary of the change and which issue is fixed.
List any external dependencies that are required for this change.

Completes: 
[AB#214551](https://dev.azure.com/Equinor/fa63ceea-8883-41e2-8823-a3b54e4be178/_workitems/edit/214551)

## Checklist:

-   [x] I have performed a self-review of my own code.
-   [x] I have linked my DevOps task using the AB# tag.
-   [x] My code is easy to read, and comments are added where needed.
-   [ ] My code is covered by tests.

### CODE REVIEW

File: src/apps/ScopeChangeRequest/Components/Form/IsWarrantyCheckbox/IsWarrantyCheckbox.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    handleInput: (val: boolean) => void;
+}
+
+export const IsWarrantyCaseCheckbox = ({
+    initialValue,
+    handleInput,
+}: IsWarrantyCaseCheckboxProps): JSX.Element => {
+    const [isChecked, setIsChecked] = useState(Boolean(initialValue));
+
+    useEffect(() => {
```
Comment (jaysencpp) [inline src/apps/ScopeChangeRequest/Components/Form/IsWarrantyCheckbox/IsWarrantyCheckbox.tsx:15]: denne skjønte jeg ikke?

Comment (Gustav-Eikaas) [reply src/apps/ScopeChangeRequest/Components/Form/IsWarrantyCheckbox/IsWarrantyCheckbox.tsx:15]: Boolean biten? initialValue kan være undefined, bare parser det til boolean. Kanskje ikke nødvendig??

Comment (Gustav-Eikaas) [reply src/apps/ScopeChangeRequest/Components/Form/IsWarrantyCheckbox/IsWarrantyCheckbox.tsx:15]: Hvis jeg tar det vekk så blir usestaten typen Boolean | undefined, istedenfor bare Boolean

File: src/apps/ScopeChangeRequest/workspaceConfig/filter/guesstimate/guesstimate.ts [hunk 2]
```diff
@@ -8,7 +8,10 @@ export const guesstimate = new Map<GuesstimateRanges, number>([
     ['>5000', 5],
 ]);
 
-export function calculateGuesstimateHoursGap(guesstimateHours: number): GuesstimateRanges {
+export function calculateGuesstimateHoursGap(
+    guesstimateHours: number | null
+): GuesstimateRanges | null {
+    if (!guesstimateHours) return null;
```
Comment (jaysencpp) [inline src/apps/ScopeChangeRequest/workspaceConfig/filter/guesstimate/guesstimate.ts:14]: hvis guersstimateHours er 0 her så vil du vel egentlig returnere '0-100'?

Comment (Gustav-Eikaas) [reply src/apps/ScopeChangeRequest/workspaceConfig/filter/guesstimate/guesstimate.ts:14]: Nei, diskuterte med Henrik og det at man estimerer 0 og ikke estimerer er ikke det samme

**Review by jaysencpp — APPROVED** [review]

**Pull request merged**